File: Task2/src/HospitalityHub.Core/DTOs/Room/CreateRoomRequest.cs
using HospitalityHub.Core.Enums;

namespace HospitalityHub.Core.DTOs.Room;

public record CreateRoomRequest(
    int Number,
    ERoomStatus Status,
    decimal BasePrice,
    decimal DiscountPercent,
    int Stage,
    ERoomType RoomType,
    int HotelId,
    string BaseLockUri,
    string ApiKey
);

File: Task4/hospitality-hub/src/app/api-proxy/models/create-room-request.ts
/* tslint:disable */
/* eslint-disable */
import { ERoomStatus } from '../models/e-room-status';
import { ERoomType } from '../models/e-room-type';
export interface CreateRoomRequest {
  apiKey?: string | null;
  baseLockUri?: string | null;
  basePrice?: number;
  discountPercent?: number;
  hotelId?: number;
  number?: number;
  roomType?: ERoomType;
  stage?: number;
  status?: ERoomStatus;
}
